### PR TITLE
Updated lucene-core, lucene-analyzers-common, and lucene-queryparser version to 8.10.0

### PR DESCRIPTION
**Pull request by imsayari404**
## Description

3 medium CVE resolved. 

**Upgradation performed :**

1. lucene-core(https://www.mend.io/vulnerability-database/WS-2021-0646) : 8.2.0 to 8.10.0
lucene-core (8.10.0) gets introduced through lucene-analyzers-common(8.10.0)
2. lucene-analyzer-common(https://www.mend.io/vulnerability-database/WS-2021-0646) : 7.7.3 to 8.10.0
3. lucene-queryparser(https://vuln.whitesourcesoftware.com/vulnerability/WS-2021-0646) : 8.2.0 to 8.10.0

## Motivation and Context

Apache Lucene through 7.x and 8.x before 8.10 is vulnerable to a denial of service. By sending a specific regular expression query, a remote attacker could exploit this vulnerability to consume all available CPU resources.

## Impact
3 medium cve got resolved.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Security Changes

* Upgrade lucene-core to 8.10.0 in response to 'WS-2021-0646 <https://www.mend.io/vulnerability-database/WS-2021-0646>' :pr:`24168`
* Upgrade lucene-queryparser to 8.10.0 in response to 'WS-2021-0646 <https://vuln.whitesourcesoftware.com/vulnerability/WS-2021-0646>' :pr:`24168`
* Upgrade lucene-analyzer-common to 8.10.0 in response to 'WS-2021-0646 <https://www.mend.io/vulnerability-database/WS-2021-0646>' :pr:`24168`

```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2004,15 +2011,27 @@
...
-                    </exclusion>
-                </exclusions>
+                <version>${dep.lucene.version}</version>
             </dependency>
 
+<!--            <dependency>-->
+<!--                <groupId>org.apache.lucene</groupId>-->
+<!--                <artifactId>lucene-core</artifactId>-->
+<!--                <version>${dep.lucene.version}</version>-->
+<!--            </dependency>-->
```
Comment (ZacBlanco) [inline pom.xml:2021]: please remove these comments

File: pom.xml [hunk 2]
```diff
@@ -2004,15 +2011,27 @@
...
+                <groupId>org.apache.lucene</groupId>
+                <artifactId>lucene-queryparser</artifactId>
+                <version>${dep.lucene.version}</version>
+            </dependency>
+
+<!--            <dependency>-->
+<!--                <groupId>org.apache.lucene</groupId>-->
+<!--                <artifactId>lucene-sandbox</artifactId>-->
+<!--                <version>${dep.lucene.version}</version>-->
+<!--            </dependency>-->
```
Comment (ZacBlanco) [inline pom.xml:2033]: please remove these commented lines

File: pom.xml [hunk 3]
```diff
@@ -2310,6 +2329,13 @@
                         <ignoredClassPatterns combine.children="append">
                             <!-- Duplicate class is being brought in by commons-io & log4j-api -->
                             <ignoredClassPattern>META-INF.versions.9.module-info</ignoredClassPattern>
+                            <!-- Ignore duplicate classes related to lucene-core and ranger-apache -->
+<!--                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.analysis.tokenattributes.CharTermAttributeImpl</ignoredClassPattern>-->
+<!--                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.document.*</ignoredClassPattern>-->
+<!--                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.index.*</ignoredClassPattern>-->
+<!--                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.search.TopFieldCollector</ignoredClassPattern>-->
+<!--                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.store.DataInput</ignoredClassPattern>-->
+<!--                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.util.*</ignoredClassPattern>-->
```
Comment (ZacBlanco) [inline pom.xml:2338]: remove please

File: presto-pinot-toolkit/pom.xml [hunk 4]
```diff
@@ -641,6 +645,12 @@
             <groupId>commons-codec</groupId>
             <artifactId>commons-codec</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.lucene</groupId>
+            <artifactId>lucene-core</artifactId>
+            <version>8.10.0</version>
```
Comment (ZacBlanco) [inline presto-pinot-toolkit/pom.xml:652]: you shouldn't need the `<version>` tag in here if the `luecene-core` dependency is defined in the root pom's `dependencyManagement` section. I would recommend adding it to the root pom's `dependencyManagement` section with the `dep.lucene.version` variable there and remove the `<version>` tag here.

**Review by ZacBlanco — CHANGES_REQUESTED** [review]
I have a concern that we are just adding an exclusion here for all of the duplicate checkers. We should prevent the duplicate dependency from existing on the classpath in the first place. I would try to modify the strategy for this PR to prevent the duplicated class patterns rather than just adding it to the ignore list everywhere.

File: presto-pinot-toolkit/pom.xml [hunk 5]
```diff
@@ -682,6 +686,9 @@
                                     <ignoredDependency>io.grpc:::</ignoredDependency>
                                     <ignoredDependency>commons-codec:commons-codec::</ignoredDependency>
                                 </ignoredDependencies>
+                                <ignoredUnusedDeclaredDependencies>
+                                    <ignoredUnusedDeclaredDependency> org.apache.lucene:lucene-core:jar:8.10.0</ignoredUnusedDeclaredDependency>
+                                </ignoredUnusedDeclaredDependencies>
```
Comment (ZacBlanco) [inline presto-pinot-toolkit/pom.xml:691]: I found this change didn't seem to be necessary to build `presto-pinot-toolkit` without errors.

```
 ./mvnw clean install -pl presto-pinot-toolkit -am -DskipUI -DskipTests
```

**Review by ZacBlanco — CHANGES_REQUESTED** [review]

File: presto-hive/pom.xml [hunk 6]
```diff
@@ -516,6 +516,9 @@
                         <ignoredDependency>org.glassfish.jersey.core:jersey-common:jar</ignoredDependency>
                         <ignoredDependency>org.eclipse.jetty:jetty-server:jar</ignoredDependency>
                     </ignoredDependencies>
+                    <ignoredUnusedDeclaredDependencies>
+                        <ignoredUnusedDeclaredDependency>org.apache.lucene:lucene-core:jar:8.10.0</ignoredUnusedDeclaredDependency>
+                    </ignoredUnusedDeclaredDependencies>
```
Comment (ZacBlanco) [inline presto-hive/pom.xml:521]: I also found this seems to be not needed? When I checked out the branch and built the entire repository with these lines removed, it worked fine.

File: pom.xml [hunk 7]
```diff
@@ -2301,6 +2308,7 @@
                         <ignoredClassPatterns combine.children="append">
                             <!-- Duplicate class is being brought in by commons-io & log4j-api -->
                             <ignoredClassPattern>META-INF.versions.9.module-info</ignoredClassPattern>
+                            <ignoredClassPattern>META-INF.versions.9.org.apache.lucene.*</ignoredClassPattern>
```
[the remark this answers is not in the document]
Comment (ZacBlanco) [reply pom.xml:2311]: After some further research, I think it would be better to keep this ignored. You can ignore this suggestion

**Comment by ZacBlanco** [comment]
Last question - 8.10 is actually still a pretty old Lucene version. It was published on [Oct 18, 2021](https://mvnrepository.com/artifact/org.apache.lucene/lucene-core/8.10.1). Have we considered moving to a release later than 8.10?

**Comment by ZacBlanco** [comment]
After offline discussion, versions >= 9.0 seem to require Java 11, so we will stick with 8.10 for now.

**Review by ZacBlanco — DISMISSED** [review]

**Review by infvg — APPROVED** [review]
LGTM

**Comment by ZacBlanco** [comment]
@tdcmeehan can you help merge this?

File: pom.xml [hunk 8]
```diff
@@ -337,6 +338,12 @@
                 <groupId>com.facebook.presto</groupId>
                 <artifactId>presto-hive</artifactId>
                 <version>${project.version}</version>
+                <exclusions>
```
Comment (pratyakshsharma) [inline pom.xml:341]: this exclusion is probably not needed. I tried building without this and the build went fine.

**Review by pratyakshsharma — CHANGES_REQUESTED** [review]
Thank you for the fix @imsayari404. Only one comment. Also please update the commit message to start with `Update` to follow imperative tone - https://cbea.ms/git-commit/#imperative

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**